### DISCUSSION TITLE
How to prevent browser default page saving dialog pop-out when clicking CTRL+ S

### DISCUSSION THREAD

**Discussion by HonTung** [GrapesJS/grapesjs#5191] [Q&A] [answered]
Hi,

I would like know to how to prevent browser default page saving dialog pop-out when clicking CTRL+ S. Below is my command code for saving function using [GrapesJS Keymap](https://grapesjs.com/docs/api/keymaps.html#add) 

```
keymaps.add('ns:save-keymap', '⌘+s, ctrl+s',  editor => {
    editor.runCommand('save-db');
});

editor.on('keymap:emit', (id, shortcut, event) => {
    switch(id){
        case 'ns:save-keymap':
            event.preventDefault();
            event.stopPropagation();
            //alert('Saving template...');
            break;
    }
});
```

In order to bypass browser default page saving dialog pop-out in GrapesJS saving command, the `event.preventDefault(); ` already put into event trigger, but seem not working. 

Anyone has any idea or experience with it?

Thank you.

**Comment by artf** [comment, marked as answer]
There is an option for that use case (I'll probably update the docs with an example)

```js
keymaps.add('ns:save-keymap', '⌘+s, ctrl+s',  editor => {
    editor.runCommand('save-db');
}, {
 prevent: true
});
```

**Reply by HonTung** [reply]
Thank you @artf it works well

**Reply by srinivasskeme** [reply]
Hey @artf, I have a similar issue with 'esc' key.

I am rendering grapesjs inside a dialog.

I want to achieve `{prevent:true}` only when some dialog / modal is open inside grapesjs. If there are no modals open within grapesjs, I would like to trigger the close of the parent modal.

**Reply by srinivasskeme** [reply]
There are events `modal:open` and `modal:close` which can be used to conditionally add prevent to esc key.
https://grapesjs.com/docs/api/modal_dialog.html#modal
